### PR TITLE
BHBC-1788: Implement xml->json transform for EML

### PR DESCRIPTION
**Pull request by anissa-agahchen**
# Overview

## Links to Jira tickets

- https://quartech.atlassian.net/browse/BHBC-1788

## Description of relevant changes

- {List all relevant changes, in particular anything that will help the reviewers test/verify this PR}

## PR Checklist

A list of items that are good to consider when making any changes.

_Note: this list is not exhaustive, and not all items are always applicable._

### Code

- [ ] New files/classes/functions have appropriately descriptive names and comment blocks to describe their use/behaviour
- [ ] I have avoided duplicating code when possible, moving re-usable pieces into functions
- [ ] I have avoided hard-coding values where possible and moved any re-usable constants to a constants file
- [ ] My code is as flat as possible (avoids deeply nested if/else blocks, promise chains, etc)
- [ ] My code changes account for null/undefined values and handle errors appropriately
- [ ] My code uses types/interfaces to help describe values/parameters/etc, help ensure type safety, and improve readability

### Style

- [ ] My code follows the established style conventions
- [ ] My code uses native material-ui components/icons/conventions when possible

### Documentation

- [ ] I have commented my code sufficiently, such that an unfamiliar developer could understand my code
- [ ] I have added/updated README's and related documentation, as needed

### Tests

- [ ] I have added/updated unit tests for any code I've added/updated
- [ ] I have added/updated the Postman requests/tests to account for any API endpoints I've added/updated

### Linting/Formatting

- [ ] I have run the linter and fixed any issues, as needed  
       _See the `lint` commands in package.json_
- [ ] I have run the formatter and fixed any issues, as needed  
       _See the `format` commands in package.json_

### SonarCloud

- [ ] I have addressed all SonarCloud Bugs, Vulnerabilities, Security Hotspots, and Code Smells

### CODE REVIEW

File: app/src/components/fields/AutocompleteField.tsx [hunk 1]
```diff
@@ -30,7 +30,7 @@ const AutocompleteField: React.FC<IAutocompleteField<string | number>> = <T exte
     const result = props.options.find((option) => existingValue === option.value);
 
     if (!result) {
-      return null as unknown as IAutocompleteFieldOption<T>;
+      return (null as unknown) as IAutocompleteFieldOption<T>;
```
Comment (NickPhura) [inline app/src/components/fields/AutocompleteField.tsx]: (not this line specifically) It looks like some formatting is changing here and in some other files in the app. I believe the most recent version of the prettier formatter actually removes these brackets, so it might be that you need to run `make install` to ensure you have the latest version and then re-run the formatter, as I think there should be no formatting related changes here.

File: api/src/services/submission-service.ts [hunk 2]
```diff
@@ -9,8 +11,12 @@ import {
   SUBMISSION_MESSAGE_TYPE,
   SUBMISSION_STATUS_TYPE
 } from '../repositories/submission-repository';
+import { getFileFromS3 } from '../utils/file-utils';
+import { getLogger } from '../utils/logger';
 import { DBService } from './db-service';
 
+const defaultLog = getLogger('paths/dwc/validate');
```
Comment (NickPhura) [inline api/src/services/submission-service.ts]: nit: update the string here.

File: api/src/services/submission-service.ts [hunk 3]
```diff
@@ -131,4 +142,84 @@ export class SubmissionService extends DBService {
   async listSubmissionRecords(): Promise<ISubmissionModelWithStatus[]> {
     return this.submissionRepository.listSubmissionRecords();
   }
+
+  /**
+   *
```
Comment (NickPhura) [inline api/src/services/submission-service.ts]: nit: add a description here

File: api/src/services/submission-service.ts [hunk 4]
```diff
@@ -131,4 +142,84 @@ export class SubmissionService extends DBService {
...
+   * @param {number} submissionId
+   * @return {*}  {Promise<ISourceTransformModel['metadata_transform_precompile']>}
+   * @memberof SubmissionService
+   */
+  async getEMLStyleSheetKey(submissionId: number): Promise<ISourceTransformModel['transform_precompile_key']> {
+    return (await this.submissionRepository.getSourceTransformIdBySubmissionId(submissionId)).transform_precompile_key;
+  }
+
+  /**
+   *
```
Comment (NickPhura) [inline api/src/services/submission-service.ts]: nit: add a description here

File: api/src/services/submission-service.ts [hunk 5]
```diff
@@ -131,4 +142,84 @@ export class SubmissionService extends DBService {
...
+   *
+   * @param {number} submissionId
+   * @return {*}  {Promise<any>}
+   * @memberof SubmissionService
+   */
+  async getStylesheetFromS3(submissionId: number): Promise<any> {
```
Comment (NickPhura) [inline api/src/services/submission-service.ts]: Lets specify the return type here, rather than saying `any`.

File: api/src/services/submission-service.ts [hunk 6]
```diff
@@ -131,4 +142,84 @@ export class SubmissionService extends DBService {
...
+      }
+
+      const s3File = await getFileFromS3(stylesheet_key);
+
+      if (!s3File) {
+        throw new HTTP500('Failed to get file from S3');
+      }
+      return s3File;
+    } catch (error) {
+      defaultLog.error({ label: 'getS3File', message: 'error', error });
```
Comment (NickPhura) [inline api/src/services/submission-service.ts]: nit: can you change `getS3File` to the name of this function

Comment (NickPhura) [reply api/src/services/submission-service.ts]: Also, now that I look at it, may as well just remove the try catch altogether since the error just gets re thrown. Could maybe argue the log message is valuable, but we'll probably end up logging something higher up as a result of this error anyways

File: app/src/hooks/api/useAxios.test.ts [hunk 7]
```diff
@@ -13,7 +13,7 @@ describe('APIError', () => {
       }
     } as Partial<AxiosError>;
 
-    const apiError = new APIError(error as unknown as AxiosError);
+    const apiError = new APIError((error as unknown) as AxiosError);
```
Comment (NickPhura) [inline app/src/hooks/api/useAxios.test.ts]: Looks like the same formatting thing I mentioned here: https://github.com/bcgov/biohubbc-platform/pull/46/files?file-filters%5B%5D=.tsx&show-deleted-files=true&show-viewed-files=true#r890750038

File: database/src/migrations/20220602152015_add_record_sims_eml_transformation.ts [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+    SET SEARCH_PATH = ${DB_SCHEMA};
+
+    ALTER TABLE survey DROP COLUMN surveyed_all_areas;
+
+    -- Create 'survey' view
+
+    SET SEARCH_PATH = ${DB_SCHEMA_DAPI_V1};
+
+    CREATE OR REPLACE VIEW survey AS SELECT * FROM ${DB_SCHEMA}.survey;
+  `);
```
Comment (NickPhura) [inline database/src/migrations/20220602152015_add_record_sims_eml_transformation.ts]: This down looks like its from some unrelated migration file

File: app/src/components/fields/AutocompleteField.tsx (hunk 1 above)
Comment (anissa-agahchen) [reply app/src/components/fields/AutocompleteField.tsx]: yeah, that did it, thanks

File: api/src/services/submission-service.ts (hunk 6 above)
Comment (anissa-agahchen) [reply api/src/services/submission-service.ts]: yes, true, we do log it higher up.

File: database/src/migrations/20220602152015_add_record_sims_eml_transformation.ts (hunk 8 above)
Comment (anissa-agahchen) [reply database/src/migrations/20220602152015_add_record_sims_eml_transformation.ts]: yup, removed now

**Review by NickPhura — APPROVED** [review]
